### PR TITLE
rationalization of front page docs and walkthrough

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Making the front page docs a little more welcoming post-transition.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
rationalization of front page docs and walkthrough
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,11 +6,34 @@
 
 ## Overview
 
-AKS-Engine leverages ARM (Azure Resource Manager) to build Kubernetes IaaS in Azure. Originally [acs-engine](https://github.com/Azure/acs-engine), the project has moved here (1) to evolve without harming its legacy non-k8s bits that other non-k8s Azure upstream projects depend upon, and (2) to better reflect its affinity with [AKS](https://docs.microsoft.com/en-us/azure/aks), Azure's managed Kubernetes service offering.
+AKS-Engine leverages ARM (Azure Resource Manager) to build Kubernetes IaaS in Azure. AKS-Engine provides convenient tooling to quickly bootstrap clusters, and is the vm provisioning implementation for [AKS](https://docs.microsoft.com/en-us/azure/aks), Azure's managed Kubernetes service offering.
```
Comment (CecileRobertMichon) [inline README.md]: `vm` ->  `VM`

aks-engine does more than that for AKS (presently some upgrade + scale implementation, provisioning of other resources such as vnets, templates, etc.) so I don't think that's an accurate description of reality.

Comment (jackfrancis) [reply README.md]: What's a better way to express in a single clause a simple summary of the abstraction layer that aks-engine fulfills for AKS?

Comment (mboersma) [reply README.md]: `implements cluster provisioning and lifecycle operations for [AKS]()...`?

File: docs/README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# Microsoft Azure Container Service Engine - Builds Docker Enabled Clusters
+# AKS-Engine - Builds Docker Enabled Clusters
```
Comment (CecileRobertMichon) [inline docs/README.md]: s/Builds Docker Enabled Clusters/Something about unmanaged Kubernetes clusters on Azure?

File: docs/README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 ## Overview
 
 This cluster definition examples demonstrate how to create a customized Docker Enabled Cluster on Microsoft Azure.
 
 ## User Guides
 
-* [AKS Engine](acsengine.md) - shows you how to build and use the AKS engine to generate custom Docker enabled container clusters
+* [AKS Engine](aksengine.md) - shows you how to build and use the AKS engine to generate custom Docker enabled container clusters
```
Comment (CecileRobertMichon) [inline docs/README.md]: Slight preference for `AKS-Engine` instead of `the AKS engine`

File: docs/design/proposed-template-refactor.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# Acs-engine
+# AKS-Engine
 
-The Azure Container Service Engine (aks-engine) is a command line tool that generates ARM (Azure Resource Manager) templates in order for one to deploy container-based clusters (like Kubernetes , DCOS, Openshift, Docker swarm) on the Azure platform.
+AKS-Engine is a command line tool that generates ARM (Azure Resource Manager) templates in order for one to deploy container-based clusters (like Kubernetes , DCOS, Openshift, Docker swarm) on the Azure platform.
```
Comment (CecileRobertMichon) [inline docs/design/proposed-template-refactor.md]: should we remove `DCOS, Openshift, Docker swarm` ?

Comment (jackfrancis) [reply docs/design/proposed-template-refactor.md]: Yes, because we will be removing that non-ACS-dependent stuff that implements those orchestrators.

Comment (mboersma) [reply docs/design/proposed-template-refactor.md]: > we will be removing that non-ACS-dependent stuff that implements those orchestrators

See #92.

File: docs/dcos.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Microsoft Azure Container Service Engine - DC/OS Walkthrough
```
Comment (mboersma) [inline docs/dcos.md]: This file has been removed in master.

File: docs/README.md (hunk 3 above)
Comment (mboersma) [reply docs/README.md]: I preferred "AKS Engine" to refer to the project and "aks-engine" for the command-line tool. But I see @jackfrancis prefers "AKS-Engine," and I'm ok with that. I just want to see consistency.

**Comment by mboersma** [comment]
/lgtm

after rebase

**Comment by CecileRobertMichon** [comment]
let's get this in before the release @jackfrancis ?

**Comment by jackfrancis** [comment]
See #152

**Pull request closed without merging**